### DISCUSSION TITLE
Automation - Split transfer into multiple Piggy banks

### DISCUSSION THREAD

**Discussion by bartekd123** [firefly-iii/firefly-iii#6116] [General]
I am currently doing the following
 - during pay-day I have a transfer from my main account into another account that I use for piggy banks (easy example, $50 from main account where I am paid into another account)
 - I have an automation that looks at the source, destination and amount, and deletes that transfer (eg Delete the transaction that matches the above)
 - i have multiple automated recurring transactions that adds the amount that adds up to the $50 (eg I have 5 different piggy banks, so on payday, I have 5 transactions for $10 each that does that same transfer as above, but it also adds to the piggy bank at the same transaction)

This does what I want, but there are a few issues with this
 - Its a workaround that has a lot of moving pieces
 - I need to have a buffer of the amount of the transaction in the piggy account, otherwise there will not be enough money in the account to run the automation if I am not able to do my import a day or 2 later
 - If the amount changes, it makes it difficult to update the various rules

What I would suggest (not sure if it is possible)
 - Have an automation that allows to split the original transaction
 - For the automation also allow the input of the money into the piggy bank

I was going to write a feature request, and you can see if it is possible but wanted to get your take on it first.

**Comment by JC5** [comment]
I have the same problem. I haven't taken the time yet to automate it. Personally I'm thinking of using webhooks and a little script. I'm not sure if I can add something to Firefly III that captures all of the use cases.

**Reply by bartekd123** [reply]
You mean you use a script that runs that splits that transaction and adds it to the various piggy banks?

Would it be a big code change to allow automations to add to piggy banks and to split? I would imagine at least the split transaction would be tough with coding since it would need to know the total amount of that transaction and how much is left over etc.

**Reply by JC5** [reply]
> Would it be a big code change to allow automations to add to piggy banks and to split? 

Yes, it would be, I'm not going to add it to Firefly III I'm afraid.

**Reply by bartekd123** [reply]
No problem at all, just wanted to suggest it. I’ll carry on the way I am. Thanks again. :)

**Reply by bartekd123** [reply]
I had another idea, that would make firefly III more customizable with automations that would help for this request. Make an automation trigger web service calls. So if an automation is triggered, it can run a list of web calls against itself or to run a script. Maybe something to consider for a future version.

**Reply by JC5** [reply]
This is already in place, it's just not in the UI (yet). At least, if this is what you mean: https://api-docs.firefly-iii.org/#/webhooks

**Reply by bartekd123** [reply]
Ooh, something new to explore. Thanks! I’ll play around with it.

**Comment by diamant-x** [comment]
Hi, Sorry for necroposting but I'm recently facing the same doubts/Scenario.
Maybe on the automation side, an action could be "split transaction up by X amount" and the input field the amount (in transaction currency) that should be allocated to the first split and Total-X should be in the second split. The splitted transactions would have the same metadata as original (i.e. same source/destination/date/description) and maybe add at the end of Description a "Split 1/2" and "Split 2/2". then a second rule would be able to update metadata as required by matching same fields + description ending in "Split x/2"

This would allow to cover a simple scenario for a single piggy bank. Potentially, setting up additional recurrent rules to run after split, one could do this endlessly number of times needed, one per piggy bank.
This would allow to also cover scenarios such as loan payments to split X amount as main capital returnal and Y amount as interests.

Open to other solutions as well.
An easier one would be to make at the bank splits per piggy bank but... feels like it shouldn't be done that way.
